src/main/java/org/datadog/jmxfetch/JMXAttribute.java

```diff
@@ -100,39 +123,35 @@ Object getJmxValue() throws AttributeNotFoundException, InstanceNotFoundExceptio
     }
 
     boolean matchDomain(Configuration conf) {
-        LinkedHashMap<String, Object> include = conf.getInclude();
-        return include.get("domain") == null || include.get("domain").equals(domain);
+        Filter include = conf.getInclude();
+        return include.getDomain() == null || include.getDomain().equals(domain);
     }
 
     boolean excludeMatchDomain(Configuration conf) {
-        LinkedHashMap<String, Object> exclude = conf.getExclude();
-        return exclude.get("domain") != null && exclude.get("domain").equals(domain);
+        Filter exclude = conf.getExclude();
+        return exclude.getDomain() != null && exclude.getDomain().equals(domain);
     }
 
     boolean excludeMatchBean(Configuration conf) {
-        LinkedHashMap<String, Object> exclude = conf.getExclude();
-        String bean = (String) exclude.get("bean");
-        String confBeanName = (String) exclude.get("bean_name");
-
-        if (beanName.equals(bean) || beanName.equals(confBeanName)) {
-            return true;
-        }
-
-        for (String bean_attr : exclude.keySet()) {
-            if (EXCLUDED_BEAN_PARAMS.contains(bean_attr)) {
-                continue;
-            }
-
-            HashMap<String, String> beanParams = new HashMap<String, String>();
-            for (String param : this.defaultTags) {
-                String[] paramSplit = param.split(":");
-                beanParams.put(new String(paramSplit[0]), new String(paramSplit[1]));
-            }
-
-            if (exclude.get(bean_attr).equals(beanParams.get(bean_attr))) {
-                return true;
-            }
-        }
+       Filter exclude = conf.getExclude();
+       ArrayList<String> beanNames = exclude.getBeanNames();
+
+       if(beanNames.contains(beanName)){
+    	   return true;
+       }
+
+       for (String bean_attr : exclude.keySet()) {
+           if (EXCLUDED_BEAN_PARAMS.contains(bean_attr)) {
+               continue;
+           }
+
+           ArrayList<String> beanValues = exclude.getParameterValues(bean_attr);
+           for (String beanVal : beanValues) {
+               if (beanParameters.get(bean_attr) != null && beanParameters.get(bean_attr).equals(beanVal)) {
```

Comment:
You should get the first condition out of the loop, and then just test the presence of the attribute in the list.